src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 5;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int MAX_BOUND = 100;
+    private static final int MIN = 1;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private String randomFigureName;
+    private Figure randomFigure;
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FigureEnumeration.values().length);
+        randomFigureName = FigureEnumeration.values()[index].name();
+        switch (randomFigureName) {
+            case "ISOSCELES_TRAPEZOID":
+                String trapezoidColor = colorSupplier.getRandomColor();
```

Comment:
you have the same line in all cases, move it before switch and use it in every case